### PR TITLE
Makeup lesson

### PR DESCRIPTION
**Pull request by bryancheny**
Fixes https://github.com/AY2425S1-CS2103T-T08-2/tp/issues/26

### CODE REVIEW

File: src/main/java/keycontacts/storage/JsonAdaptedMakeupLesson.java [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+
+    /**
+     * Constructs a {@code JsonAdaptedMakeupLesson} with no arguments. This is used by Jackson to create a new
+     * {@code JsonAdaptedMakeupLesson} object.
+     */
+    public JsonAdaptedMakeupLesson() {
+        this.lessonDate = "";
+        this.startTime = "";
+        this.endTime = "";
+    }
```
Comment (IzN432) [inline src/main/java/keycontacts/storage/JsonAdaptedMakeupLesson.java:28]: Why is this constructor needed?

Comment (bryancheny) [reply src/main/java/keycontacts/storage/JsonAdaptedMakeupLesson.java:28]: The tests won't compile if this default constructor is not here

File: src/main/java/keycontacts/logic/commands/MakeupLessonCommand.java [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+    private final MakeupLesson makeupLesson;
+
+    /**
+     * @param date      of the makeup lesson
+     * @param startTime of the makeup lesson
+     * @param endTime   of the makeup lesson
+     */
+    public MakeupLessonCommand(Index targetIndex, MakeupLesson makeupLesson) {
+        this.targetIndex = targetIndex;
+        this.makeupLesson = makeupLesson;
```
Comment (IzN432) [inline src/main/java/keycontacts/logic/commands/MakeupLessonCommand.java:49]: add `requireAllNonNull(targetIndex, makeupLesson);`

File: src/main/java/keycontacts/storage/JsonAdaptedMakeupLesson.java (hunk 1 above)
Comment (dwsc37) [reply src/main/java/keycontacts/storage/JsonAdaptedMakeupLesson.java:28]: This should not be the case. Probably the sign of some underlying error

File: docs/UserGuide.md [hunk 3]
```diff
@@ -218,6 +232,7 @@ Action     | Format, Examples
 **Edit**   | `edit INDEX [n/NAME] [p/PHONE_NUMBER] [a/ADDRESS]`<br> e.g.,`edit 2 n/James Lee p/81234567`
 **Assign** | `assign INDEX pn/PIECE_NAME...`<br> e.g,`assign 1 pn/Moonlight Sonata pn/Canon in D`
 **Schedule** | `schedule INDEX d/DAY st/START_TIME et/END_TIME`<br> e.g.,`schedule 1 d/Monday st/12:00 et/14:00`
+**Makeup** | `makeup INDEX dt/DATE st/START_TIME et/END_TIME`<br> e.g.,`makeup 1 d/2022-12-25 st/12:00 et/14:00`
```
Comment (IzN432) [inline docs/UserGuide.md:235]: change date format here and above

File: src/main/java/keycontacts/storage/JsonAdaptedMakeupLesson.java (hunk 1 above)
Comment (bryancheny) [reply src/main/java/keycontacts/storage/JsonAdaptedMakeupLesson.java]: ok, removed!

File: src/main/java/keycontacts/logic/commands/MakeupLessonCommand.java (hunk 2 above)
Comment (bryancheny) [reply src/main/java/keycontacts/logic/commands/MakeupLessonCommand.java:51]: done!

File: docs/UserGuide.md (hunk 3 above)
Comment (bryancheny) [reply docs/UserGuide.md]: Done

**Review by dwsc37 — APPROVED** [review]
LGTM

**Pull request merged**